subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/GradlePomModuleDescriptorParser.java

```diff
@@ -204,10 +209,20 @@ private PomReader parseParentPom(DescriptorParseContext parseContext, ModuleComp
     }
 
     private PomReader parsePom(DescriptorParseContext parseContext, ModuleComponentIdentifier parentId, Map<String, String> childProperties) throws IOException, SAXException {
-        LocallyAvailableExternalResource localResource = parseContext.getMetaDataArtifact(parentId, ArtifactType.MAVEN_POM);
+        LocallyAvailableExternalResource localResource = getMetaDataArtifact(parseContext, parentId);
         PomReader pomReader = new PomReader(localResource, moduleIdentifierFactory, childProperties);
         GradlePomModuleDescriptorBuilder mdBuilder = new GradlePomModuleDescriptorBuilder(pomReader, gradleVersionSelectorScheme, mavenVersionSelectorScheme, moduleIdentifierFactory, moduleExclusions);
         doParsePom(parseContext, mdBuilder, pomReader);
         return pomReader;
     }
+
+    private LocallyAvailableExternalResource getMetaDataArtifact(DescriptorParseContext parseContext, ModuleComponentIdentifier parentId) {
+        if (mavenVersionSelectorScheme.parseSelector(parentId.getVersion()).isDynamic()) {
```

Comment:
We shouldn't be doing this. If the input is a _selector_, meaning it _might_ have a dynamic version, then we should treat it as such. Don't even check if the version looks like it's dynamic. We either have a `ComponentIdentifier` (in which case we know exactly the component to use, or a `ComponentSelector` in which case we do some logic to get the identifier. Part of this logic is checking for a dynamic version: but this is done inside the dependency resolution engine.

In summary: if the input is a selector (might have a dynamic version), then we should treat it like it is dynamic and let the dependency resolution engine to the conversion.